src/server/pkg/sync/sync.go

```diff
@@ -38,14 +38,23 @@ func NewPuller() *Puller {
 // fileInfo is the file/dir we are puuling.
 // pipes causes the function to create named pipes in place of files, thus
 // lazily downloading the data as it's needed.
-func (p *Puller) Pull(client *pachclient.APIClient, root string, repo, commit, file string, pipes bool, concurrency int) error {
+func (p *Puller) Pull(client *pachclient.APIClient, root string, repo, commit, file string, pipes bool, concurrency int, preservePrefix bool) error {
```

Comment:
How would you feel about not having the `preservePrefix` argument and just implementing only the `false` behavior. Callers can preserve whatever prefix they'd like by passing that prefix as part of `root`, seems a little bit cleaner to me to only have 1 way of doing it.